src/elements/content-sharing/SharingNotification.js

```diff
@@ -167,6 +179,66 @@ function SharingNotification({
         setSharedLink,
     ]);
 
+    /**
+     * Get the item's collaborators
+     *
+     * A note on the wording: the USM uses the term "collaborators" internally,
+     * so the state variable and state setting function also refer to "collaborators."
+     * However, we are using the Collaborations API here, so the API-related functions
+     * use the term "collaborations." For more details, see ./api/FileCollaborations.
+     */
+    React.useEffect(() => {
+        const handleGetCollaborationsSuccess = (response: Collaborations) => {
+            const updatedCollaboratorsList = convertCollabsResponse(response, ownerEmail, ownerID === currentUserID);
+            setCollaboratorsList(updatedCollaboratorsList);
+            setRetrievedCollaborators(true);
+        };
+
+        const handleGetCollaborationsError = () => {
+            const updatedNotifications = { ...notifications };
+            if (updatedNotifications[notificationID]) {
+                return;
+            }
+            updatedNotifications[notificationID] = createNotification(
+                STATUS_ERROR,
+                contentSharingMessages.collaboratorsLoadingError,
+            );
+            setCollaboratorsList({ collaborators: [] }); // default to an empty collaborators list for the USM
+            setNotifications(updatedNotifications);
+            setNotificationID(notificationID + 1);
+            setRetrievedCollaborators(true);
+        };
+
+        if (itemID && currentUserID && !retrievedCollaborators) {
+            let collabAPIInstance;
+            if (itemType === TYPE_FILE) {
+                collabAPIInstance = api.getFileCollaborationsAPI(false);
+            } else if (itemType === TYPE_FOLDER) {
+                collabAPIInstance = api.getFolderCollaborationsAPI(false);
+            }
+            if (collabAPIInstance) {
+                collabAPIInstance.getCollaborations(
+                    itemID,
+                    handleGetCollaborationsSuccess,
+                    handleGetCollaborationsError,
+                );
+            }
+        }
+    }, [
+        api,
+        collaboratorsList,
+        createNotification,
+        currentUserID,
+        itemID,
+        itemType,
+        notificationID,
+        notifications,
+        ownerEmail,
+        ownerID,
+        retrievedCollaborators,
+        setCollaboratorsList,
```

Comment:
Does this effect need to re-run based on all of these values?

Reply:
Yup, this is actually an optimization. React will only run this effect if any of the variables in the dependency array has changed; otherwise, it will skip the effect. (Also, the React linter throws an error if I try to remove any of the dependencies)